docs/dev/testing.rst

```diff
@@ -0,0 +1,263 @@
+passed through to ``pytest``. We've configured ``pytest`` (through the
+``test/conftest.py`` configuration file) to be on the lookout for the
+``--tmp-data`` flag and act accordingly.
+
+.. seealso::
+
+    * :doc:`/usage` for more on how to set up a PUDL workspace, including a
+      datastore.
+    * :doc:`/datastore` for more on how to work with the datastore.
+
+-------------------------------------------------------------------------------
+Running pytest Directly
+-------------------------------------------------------------------------------
+Running tests directly with ``pytest`` gives you the ability to run only
+tests from a particular test module, or even a single individual test case.
+It's also faster because there's no testing environment to set up. Instead,
+it just uses your Python environment, which should be the ``pudl-dev`` conda
+environment discussed in :doc:`/dev/dev_setup`. This is convenient if you're
+debugging something specific, or developing new test cases, but it's not as
+robust as using Tox.
+
+Running specific tests
+^^^^^^^^^^^^^^^^^^^^^^
+To run the software unit tests with ``pytest`` directly:
+
+.. code-block:: console
+
+   $ pytest test/unit
+
+To run only the unit tests for the Excel spreadsheet extraction module:
+
+.. code-block:: console
+
+   $ pytest test/unit/extract/excel_test.py
+
+To run only the unit tests defined by a single test class within that module:
+
+.. code-block:: console
+
+   $ pytest test/unit/extract/excel_test.py::TestGenericExtractor
+
+Custom PUDL pytest flags
+^^^^^^^^^^^^^^^^^^^^^^^^
+We have defined several custom flags to control pytest's behavior when running
+the PUDL tests. You can always check to see what custom flags exist by running
+``pytest --help`` and looking at the ``custom options`` section:
+
+.. code-block:: console
+
+  custom options:
+  --live-dbs            Use existing PUDL/FERC1 DBs instead of creating temporary ones.
+  --tmp-data            Download fresh input data for use with this test run only.
+  --etl-settings=ETL_SETTINGS
+                        Path to a non-standard ETL settings file to use.
+  --gcs-cache-path=GCS_CACHE_PATH
+                        If set, use this GCS path as a datastore cache layer.
+  --sandbox             Use raw inputs from the Zenodo sandbox server.
+
+Run the output tests using a pre-existing PUDL database rather than building
+a new one for testing:
+
+.. code-block:: console
+
+  $ pytest --live-dbs test/integration/fast_output_test.py
+
+Run the ETL portion of the integration tests, and force it to download fresh
+input data to a temporary datastore:
+
+.. code-block:: console
+
+   $ pytest --tmp-data test/integration/etl_test.py
+
+Run the ETL portion of the integration tests, but using a non-standard settings
+file (equivalent to ``tox -e ferc1_solo``):
+
+.. code-block:: console
+
+  $ pytest --etl-settings=test/settings/ferc1-solo-test.yml test/integration/etl_test.py
```

Comment:
slightly confusing to have duplicate information here. What is the benefit of running in in tox or pytest? What would be an example of when you might use a non-standard settings file. Or rather, how would you define a non-standard settings file?

Reply:
Okay I've re-written to try and make this clearer.